### ISSUE TITLE
Customizable Issue title display format

### ISSUE THREAD

**Issue by domtronn** [domtronn/magit-gh-issues.el#54]
It would be nice if the users were able to decide the formatting that they would like for the issue titles, currently there's predefined format of

```
%ID% %COMMENTS% %AVATAR% %TITLE% %LABELS%
```

If the user could write a string _similar_ to the above, then we get the ability to toggle out certain things implicitly, _i.e._ if the user does not include `"%AVATAR%"` then there's no need to fetch the img _(performance bonus! Although I would probably still have that as another variable)_.

My thinking is that I create an `alist` mapping between the format objects and the functions which create them, _e.g._

``` lisp
'((id . 'magit-gh-issues--make-id-string)
  (comments . 'magit-gh-issues--make-comments-string)
  (avatar . 'magit-gh-issues--get-avatar)
  ...)
```

And just map the elements of the _format_ string to these functions.

The difficulty comes with calling these _with arguments_ and the notion of _pre_ and _post_ avatar insertion. Since the users avatar has to be inserted using native `insert`, I'd have to push the `magit-insert` calls back into each of `-make-thing-string` functions.

**Comment by domtronn** [comment]
To address the applying arguments to the functions, I could make each of them take the same API and pick the things they want out of that?

**Comment by domtronn** [comment]
This has now been implemented in 0.3.0 onwards
